cmd/release-controller/upgrades.go

```diff
@@ -291,7 +292,7 @@ func (g *UpgradeGraph) Load(r io.Reader) error {
 	return err
 }
 
-func syncGraphToSecret(graph *UpgradeGraph, update bool, secretClient kv1core.SecretInterface, ns, name string, stopCh <-chan struct{}) {
+func syncGraphToSecret(graph *UpgradeGraph, update bool, secretClient kv1core.SecretInterface, ns, name string, stopCh <-chan struct{}, c *Controller) {
```

Comment:
Generally this would not be allowed.  You don't pass massive top level objects, you'd pass an interface

```
type GraphPruner interface {
  // Godoc that says what i do
  PruneGraph()
}
```

and 

```
func syncGraphToSecret(..., pruner GraphPruner) {
  ...
  if pruner != nil {
    pruner.PruneGraph()
  }
}
```